client/allochealth/tracker.go

```diff
@@ -387,6 +391,13 @@ OUTER:
 			}
 		}
 
+		if foundChecks < t.consulCheckCount {
+			// didn't see all the checks we expect, task might be slow to start
+			// or restarting tasks that had their service hooks deregistered
+			passed = false
```

Comment:
Will this logic work when it is a migration for a batch job: https://github.com/hashicorp/nomad/blob/c346a47b5b3db306c60a8a598f20dd13c5599db9/client/allocrunner/health_hook.go#L253?

Reply:
I believe so - the assumption this depends on that checks are unique and that job definition has a 1 to 1 mapping with consul agent check registration, regardless of exact parameter or minHealthyTime.  This needs to be tweaked for task group level checks, but will address that in a follow up PR.

Reply:
Sorry should have added more for my thinking. If it is a batch job, the tasks can finish and then the number of checks that will be registered in Consul will be less than those defined in the job spec.

Reply:
Ah - excellent point.  Going to drawing board a bit to think this through a bit more.  Not sure if I fully understand expectations of service health checks in batch - but it's definitely a problem for service jobs with init containers with task lifecycle/dependencies features.  Thanks for raising it.

Reply:
Checked - and looks like batch jobs don't support migrate or update/deployments [1] - do you know when this code is relevant for batch by any chance?

[1]

Reply:
Ah sorry for the noise! Thought batch supported migrate!

Reply:
:) You helped found a bug though - addressed it with a different approach in https://github.com/hashicorp/nomad/pull/7383 .